ctapipe/tools/train_energy_regressor.py

```diff
@@ -113,31 +122,56 @@ def start(self):
             self.log.info("done")
 
     def _read_table(self, telescope_type):
-        table = self.loader.read_telescope_events([telescope_type])
-        self.log.info("Events read from input: %d", len(table))
-        if len(table) == 0:
-            raise TooFewEvents(
-                f"Input file does not contain any events for telescope type {telescope_type}"
-            )
+        chunk_iterator = self.loader.read_telescope_events_chunked(
+            self.chunk_size,
+            telescopes=[telescope_type],
+        )
+        bar = tqdm(
+            chunk_iterator,
+            desc=f"Loading training events for {telescope_type}",
+            unit=" Array Events",
+            total=chunk_iterator.n_total,
+        )
+        table = []
+        n_events_in_file = 0
+        n_valid_events_in_file = 0
+
+        with bar:
+            for chunk, (start, stop, table_chunk) in enumerate(chunk_iterator):
+                self.log.debug("Events read from chunk %d: %d", chunk, len(table_chunk))
+                n_events_in_file += len(table_chunk)
+
+                mask = self.regressor.quality_query.get_table_mask(table_chunk)
+                table_chunk = table_chunk[mask]
+                self.log.debug(
+                    "Events in chunk %d after applying quality_query: %d",
+                    chunk,
+                    len(table_chunk),
+                )
+                n_valid_events_in_file += len(table_chunk)
+
+                table_chunk = self.regressor.feature_generator(
+                    table_chunk, subarray=self.loader.subarray
+                )
+                feature_names = self.regressor.features + [self.regressor.target]
+                table_chunk = table_chunk[feature_names]
+
+                valid = check_valid_rows(table_chunk)
```

Comment:
Keep track of how many and only report once at the end?

Reply:
I think doing the validity check itself after merging the chunks, is even simpler.

Reply:
The quality criteria and the column selection would still happen within the loop, but in case there are other non-predictable events left, it would be better to also drop those within the loop, I guess. I will change it.